test/worker_versioning_test.go

```diff
@@ -800,6 +801,75 @@ func (ts *WorkerVersioningTestSuite) TestReachabilityVersionsWithRules() {
 	ts.Equal(client.BuildIDTaskReachability(client.BuildIDTaskReachabilityReachable), taskQueueVersionInfo.TaskReachability)
 }
 
+func (ts *WorkerVersioningTestSuite) TestTaskQueueStats() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	fetchAndValidateStats := func(expectedWorkflowStats *client.TaskQueueStats, expectedActivityStats *client.TaskQueueStats) {
+		taskQueueInfo, err := ts.client.DescribeTaskQueueEnhanced(ctx, client.DescribeTaskQueueEnhancedOptions{
+			TaskQueue: ts.taskQueueName,
+			TaskQueueTypes: []client.TaskQueueType{
+				client.TaskQueueTypeWorkflow,
+				client.TaskQueueTypeActivity,
+			},
+			ReportStats: true,
+		})
+		ts.NoError(err)
+		ts.Equal(1, len(taskQueueInfo.VersionsInfo))
+		ts.validateTaskQueueStats(expectedWorkflowStats, taskQueueInfo.VersionsInfo[""].TypesInfo[client.TaskQueueTypeWorkflow].Stats)
+		ts.validateTaskQueueStats(expectedActivityStats, taskQueueInfo.VersionsInfo[""].TypesInfo[client.TaskQueueTypeActivity].Stats)
+	}
+
+	// Basic workflow runs two activities
+	handle, err := ts.client.ExecuteWorkflow(ctx, ts.startWorkflowOptions("basic-wf"), ts.workflows.Basic)
+	ts.NoError(err)
+
+	// Give time for the task to be sent to the TQ
+	time.Sleep(500 * time.Millisecond)
```

Comment:
We should try to avoid sleep in tests, it is very flaky, we should wait for the condition to become true

Reply:
replaced it with eventually.